### PR TITLE
Add Java bindings for t-digest reduction

### PR DESCRIPTION
**Pull request by andygrove**
Adds JNI bindings for t-digest reductions (in commit https://github.com/rapidsai/cudf/commit/756ed209427277406d75fd1a43281b88e9c82914)

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 1]
```diff
@@ -4148,6 +4148,30 @@ void testMergeApproxPercentile2() {
     }
   }
 
+  @Test
+  void testCreateTDigestReduction() {
```
[the remark this answers is not in the document]
Comment (andygrove) [reply java/src/test/java/ai/rapids/cudf/TableTest.java:4152]: Yes, there should. I'm working on adding one now.

[the remark this answers is not in the document]
Comment (andygrove) [reply java/src/test/java/ai/rapids/cudf/TableTest.java:4153]: The test is pushed but will fail until https://github.com/rapidsai/cudf/pull/10506 is merged so I added the do not merge label.

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 2]
```diff
@@ -4148,6 +4148,93 @@ void testMergeApproxPercentile2() {
...
+                    2.0), // max
+            new StructData(Arrays.asList(
+                    new StructData(3.0, 200.0),
+                    new StructData(4.0, 99.0)),
+                    3.0, // min
+                    4.0)); // max
+        Scalar merged = tdigests.reduce(ReductionAggregation.mergeTDigest(1000), DType.STRUCT)) {
+
+      assertEquals(DType.STRUCT, merged.getType());
+      ColumnView[] columns = merged.getChildrenFromStructScalar();
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/TableTest.java:4202]: The views in `columns` need to be closed or a native host memory leak occurs. `ClosableArray` might be useful here, e.g.:
```java
  try (CloseableArray views = CloseableArray.wrap(merged.getChildrenFromStructScalar()) {
``

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 3]
```diff
@@ -4148,6 +4148,93 @@ void testMergeApproxPercentile2() {
...
 
+  @Test
+  void testCreateTDigestReduction() {
+    try (Table t1 = new Table.TestBuilder()
+            .column(100, 150, 160, 70, 110, 160)
+            .build();
+         Scalar tdigest = t1.getColumn(0)
+            .reduce(ReductionAggregation.createTDigest(1000), DType.STRUCT)) {
+      assertEquals(DType.STRUCT, tdigest.getType());
+      ColumnView[] columns = tdigest.getChildrenFromStructScalar();
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/TableTest.java:4159]: Same need-to-close-views issue here.

**Review by jlowe — APPROVED** [review]

**Pull request merged**